dash-pipeline/SAI/src/dashsai.cpp

```diff
@@ -1194,29 +1395,63 @@ sai_status_t DashSai::get(
     DASH_LOG_ENTER();
     DASH_CHECK_API_INITIALIZED();
 
-    if (grpc::StatusCode::OK != readTableEntry(matchActionEntry)) {
+    if (grpc::StatusCode::OK != readTableEntry(matchActionEntry))
+    {
         return SAI_STATUS_FAILURE;
     }
 
     auto action = matchActionEntry->mutable_action()->mutable_action();
     pi_p4_id_t action_id = action->action_id();
 
-    for (uint32_t i = 0; i < attr_count; i++) {
-        if (auto meta_param = meta_table.get_meta_action_param(action_id, attr_list[i].id)) {
+    for (uint32_t i = 0; i < attr_count; i++)
+    {
+        if (auto meta_param = meta_table.get_meta_action_param(action_id, attr_list[i].id))
+        {
             // attr in table action params
-            auto pair_param = get_action_param_pair_from_p4_table_entry(meta_param, matchActionEntry);
-            if (pair_param.second) {
+            auto pair_param = get_action_param_with_is_v6_flag_from_p4_table_entry(meta_param, matchActionEntry);
+            if (pair_param.second)
+            {
                 get_attr_ipaddr_family_from_p4(pair_param.second, attr_list[i].value);
             }
 
             assert(pair_param.first);
             get_attr_value_from_p4(meta_param->field, meta_param->bitwidth, pair_param.first, attr_list[i].value);
         }
-        else {
-            DASH_LOG_ERROR("failed to get value for attr %d", attr_list[i].id);
+        else
+        {
+            get_attr_from_p4_misc(meta_table, matchActionEntry, &attr_list[i]);
         }
     }
 
     return SAI_STATUS_SUCCESS;
 }
 
+void DashSai::mutateSiblingTablesEntry(
+        _In_ const P4MetaTable &meta_table,
+        _In_ std::shared_ptr<p4::v1::TableEntry> matchActionEntry,
+        _In_ p4::v1::Update_Type updateType,
+        _In_ uint32_t action_id)
+{
+    if (meta_table.sibling_tables.empty())
+    {
+        return;
+    }
+
+    std::shared_ptr<p4::v1::TableEntry> entry = std::make_shared<p4::v1::TableEntry>();
+    entry->CopyFrom(*matchActionEntry);
```

Comment:
will be better to add a comment here to indicate all key and attribute index/id are the same.

Reply:
only the action is different.

Reply:
Added comments